internal/util/terraform.go

```diff
@@ -174,9 +197,9 @@ func (t *TerraformBinUtil) Init(ctx context.Context, args []string) error {
 }
 
 // Apply will call `terraform apply` with the given vars.
-func (t *TerraformBinUtil) Apply(ctx context.Context, tfVars []string) error {
+func (t *TerraformBinUtil) Apply(ctx context.Context, args []string) error {
 	cfg := ctx.Value(constants.DeployConfig).(*configs.DeployConfig)
-	logFile, err := t.FileSystem.OpenFileWriter(ctx.Value("deployLogFile").(string))
+	logFile, err := t.FileSystem.OpenFileWriter(ctx.Value(constants.DeployLogFile).(string))
 
 	if err != nil {
 		logFile = nil
```

Comment:
What happens here, if we fail to open the log file? Will this cause a nil-pointer reference in the command execution? Or just not log anything?

Might be worth at least logging a warning message about this, if it happens

Reply:
If the writer passed into the execCommand() method is nil, the uyse is warned and STDOUT and STDERR are used. No, you didn't miss that--I added that as a response to this comment.